src/permissions/Caveat.ts

```diff
@@ -222,6 +235,18 @@ export type ExtractCaveatValue<
   CaveatType extends string,
 > = ExtractCaveat<CaveatSpecifications, CaveatType>['value'];
 
+/**
+ * Determines whether a caveat specification has a decorator.
+ *
+ * @param specification - The caveat specification.
+ * @returns True if the caveat specification has a decorator, otherwise false.
+ */
+export function hasDecorator(
+  specification: CaveatSpecificationConstraint,
+): specification is RestrictedMethodCaveatSpecificationConstraint {
+  return 'decorator' in specification;
```

Comment:
This should use the `hasProperty` utility instead. (Currently defined in a local file, it will be replaced by the `@metamask/utils` version whenever we migrate to that.)